tests/utests/types/int8.c

```diff
@@ -0,0 +1,1758 @@
+    assert_non_null(mod);
+    lysc_leaf = (void *) mod->compiled->data;
+    CHECK_LYSC_NODE_LEAF(lysc_leaf, NULL, 0, 0x5, 1, "port", 0, 0, 0, NULL, 0, 0, NULL, NULL);
+    CHECK_LYSC_TYPE_NUM((struct lysc_type_num *)lysc_leaf->type, LY_TYPE_INT8, 0, 1);
+    range = ((struct lysc_type_num *)lysc_leaf->type)->range;
+    CHECK_LYSC_RANGE(range, NULL, NULL, NULL, 0, 1, NULL);
+    assert_int_equal(range->parts[0].min_64, 1);
+    assert_int_equal(range->parts[0].max_64, 1);
+    lysp_leaf = (void *) mod->parsed->data;
+    CHECK_LYSP_NODE_LEAF(lysp_leaf, NULL, 0, 0x0, 0, "port", 0, 0, NULL, 0, 0, NULL, NULL);
+    CHECK_LYSP_TYPE(&(lysp_leaf->type), 0, 0, 0, 0, 0, 0x80, 0, 0, "int8", 0, 0, 1, 1, 0, 0);
+    CHECK_LYSP_RESTR(lysp_leaf->type.range, "1 .. 1", NULL, NULL, NULL, 0, NULL);
+
+    //TEST MODULE T4
+    schema = MODULE_CREATE_YANG("T5", "leaf port {type int8 {range \"7\";}}");
+    UTEST_ADD_MODULE(schema, LYS_IN_YANG, NULL, LY_SUCCESS, &mod);
+    assert_non_null(mod);
+    lysc_leaf = (void *) mod->compiled->data;
+    CHECK_LYSC_NODE_LEAF(lysc_leaf, NULL, 0, 0x5, 1, "port", 0, 0, 0, NULL, 0, 0, NULL, NULL);
+    CHECK_LYSC_TYPE_NUM((struct lysc_type_num *)lysc_leaf->type, LY_TYPE_INT8, 0, 1);
+    range = ((struct lysc_type_num *)lysc_leaf->type)->range;
+    CHECK_LYSC_RANGE(range, NULL, NULL, NULL, 0, 1, NULL);
+    assert_int_equal(range->parts[0].min_64, 7);
+    assert_int_equal(range->parts[0].max_64, 7);
+    lysp_leaf = (void *) mod->parsed->data;
+    CHECK_LYSP_NODE_LEAF(lysp_leaf, NULL, 0, 0x0, 0, "port", 0, 0, NULL, 0, 0, NULL, NULL);
+    CHECK_LYSP_TYPE(&(lysp_leaf->type), 0, 0, 0, 0, 0, 0x80, 0, 0, "int8", 0, 0, 1, 1, 0, 0);
+    CHECK_LYSP_RESTR(lysp_leaf->type.range, "7", NULL, NULL, NULL, 0, NULL);
+
+    //TEST MODULE T4
+    schema = MODULE_CREATE_YANG("T6", "leaf port {type int8 {range \"min .. max\";}}");
+    UTEST_ADD_MODULE(schema, LYS_IN_YANG, NULL, LY_SUCCESS, &mod);
+    assert_non_null(mod);
+    lysc_leaf = (void *) mod->compiled->data;
+    CHECK_LYSC_NODE_LEAF(lysc_leaf, NULL, 0, 0x5, 1, "port", 0, 0, 0, NULL, 0, 0, NULL, NULL);
+    CHECK_LYSC_TYPE_NUM((struct lysc_type_num *)lysc_leaf->type, LY_TYPE_INT8, 0, 1);
+    range = ((struct lysc_type_num *)lysc_leaf->type)->range;
+    CHECK_LYSC_RANGE(range, NULL, NULL, NULL, 0, 1, NULL);
+    assert_int_equal(range->parts[0].min_64, -128);
+    assert_int_equal(range->parts[0].max_64, 127);
+    lysp_leaf = (void *) mod->parsed->data;
+    CHECK_LYSP_NODE_LEAF(lysp_leaf, NULL, 0, 0x0, 0, "port", 0, 0, NULL, 0, 0, NULL, NULL);
+    CHECK_LYSP_TYPE(&(lysp_leaf->type), 0, 0, 0, 0, 0, 0x80, 0, 0, "int8", 0, 0, 1, 1, 0, 0);
+    CHECK_LYSP_RESTR(lysp_leaf->type.range, "min .. max", NULL, NULL, NULL, 0, NULL);
+
+
+    //TEST ERROR -60 .. 0 | 0 .. 127
+    schema = MODULE_CREATE_YANG("ERR0", "leaf port {type int8 {range \"-60 .. 0 | 0 .. 127\";}}");
+    UTEST_ADD_MODULE(schema, LYS_IN_YANG, NULL, LY_EVALID, &mod);
```

Comment:
You are not adding a module but using `UTEST_ADD_MODULE` macro? Instead of rewriting the macro used everywhere, write a new one that will expect the module to fail to load.